src/frontend/apps/lti_site/components/VideoWidgetProvider/widgets/DownloadVideo/index.tsx

```diff
@@ -69,6 +94,61 @@ export const DownloadVideo = () => {
     [intl, messages],
   );
 
+  const [toggleAllowDownload, setToggleAllowDownload] = useState(
+    video.show_download,
+  );
+
+  const onSuccessCallback = useCallback(
+    (videoUpdated: Video) => {
+      toast.success(
+        intl.formatMessage(
+          videoUpdated.show_download
+            ? messages.allowDownloadToggleSuccess
+            : messages.disallowDownloadToggleSuccess,
+        ),
+        {
+          position: 'bottom-center',
+        },
+      );
+    },
+    [
+      toast,
+      intl,
+      messages.disallowDownloadToggleSuccess,
+      messages.allowDownloadToggleSuccess,
+    ],
+  );
+
+  const onErrorCallback = useCallback(
```

Comment:
Can you explain me why it is interesting in that case to use `useCallback` instead of directly defining the callback in the `onError` property of `useUpdateVideo` ?

Reply:
Dans l'idée, cest pour eviter de muter au maximum la `const videoMutation`, qui est une dépendance de `onToggleChange`, qui ferait donc rerender le component `ToggleInput`.
Apres dans les fait, cest principalement des contantes dans les useCallback, sauf `video.show_download`, donc oui, cest peut-etre pas necessaire.